### PR TITLE
git-undo: add page

### PR DESCRIPTION
**Pull request by CleanMachine1**
- [x] The page (if new), does not already exist in the repo.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

For #5137

### CODE REVIEW

File: pages/common/git-undo.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+> Undo recent commits.
+> Part of `git-extras`.
+> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-undo>.
+
+- Remove most recent commit:
+
+`git undo`
+
+- Remove a specific number of recent commits:
```
Comment (CleanMachine1) [inline pages/common/git-undo.md:11]:
```suggestion
- Remove a specific number of most recent commits:
```

Comment (CleanMachine1) [reply pages/common/git-undo.md:11]: @bl-ue opinion?

Comment (bl-ue) [reply pages/common/git-undo.md:11]: Yep, I like that.  I'd add `the` before `most` in this and the other example, though.

File: pages/common/git-undo.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# git undo
+
+> Undo recent commits.
+> Part of `git-extras`.
+> More information: <https://github.com/tj/git-extras/blob/master/Commands.md#git-undo>.
+
+- Remove most recent commit:
```
Comment (CleanMachine1) [inline pages/common/git-undo.md:7]:
```suggestion
- Remove the most recent commit:
```

**Review by bl-ue — APPROVED** [review]

**Comment by CleanMachine1** [comment]
Just need a little help. Should this be rebased when we push.

Disregard that future changes could happen to this.

**Comment by bl-ue** [comment]
No, squash.  When a pull request should be squashed you'll see a comment from an experienced maintainer, for example @waldyrious, @sbrl, @owenvoke, or @mebeim.

Rule is: default to squash unless someone thinks otherwise.

**Comment by CleanMachine1** [comment]
Ok thanks.

**Review by marchersimon — APPROVED** [review]
Interesting. Is `git undo n` the same as `git reset --hard HEAD~n`?

**Comment by CleanMachine1** [comment]
I don't know but a lot of these are just git commands with a new name

**Comment by CleanMachine1** [comment]
Commiting since I have reviewed it and seen no more ways it could be improved and also it is so small and has had 2 reviewers already

**Comment by bl-ue** [comment]
@marchersimon it seems to be the same as `git reset`_`--soft`_`HEAD~n`.  `-h` can change that, though.

**Pull request merged**